restapi/src/main/java/io/stargate/web/docsapi/service/DocsShredder.java

```diff
@@ -102,97 +120,16 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
-                            ErrorCode.DOCS_API_GENERAL_ARRAY_LENGTH_EXCEEDED);
-                      }
-
-                      // left-pad the array element to 6 characters
-                      pv = "[" + DocsApiUtils.leftPadTo6(innerPath) + "]";
-                    } else if (i == path.size()) {
-                      firstLevelKeys.add(innerPath);
-                      pv = innerPath;
-                    } else {
-                      pv = innerPath;
-                    }
-
-                    bindMap.put("p" + i++, pv);
-                    leaf = pv;
-                  }
-
-                  bindMap.put("leaf", leaf);
-
-                  if (v.isValueNode() && !v.isNull()) {
-                    ValueNode value = (ValueNode) v;
-
-                    if (value.isNumber()) {
-                      bindMap.put("dbl_value", value.asDouble());
-                      bindMap.put("bool_value", null);
-                      bindMap.put("text_value", null);
-                    } else if (value.isBoolean()) {
-                      bindMap.put("dbl_value", null);
-                      bindMap.put(
-                          "bool_value",
-                          convertToBackendBooleanValue(
-                              value.asBoolean(), db.treatBooleansAsNumeric()));
-                      bindMap.put("text_value", null);
-                    } else {
-                      bindMap.put("dbl_value", null);
-                      bindMap.put("bool_value", null);
-                      bindMap.put("text_value", value.asText());
-                    }
-                  } else if (isEmptyObject(v)) {
-                    bindMap.put("dbl_value", null);
-                    bindMap.put("bool_value", null);
-                    bindMap.put("text_value", DocumentDB.EMPTY_OBJECT_MARKER);
-                  } else if (isEmptyArray(v)) {
-                    bindMap.put("dbl_value", null);
-                    bindMap.put("bool_value", null);
-                    bindMap.put("text_value", DocumentDB.EMPTY_ARRAY_MARKER);
-                  } else {
-                    bindMap.put("dbl_value", null);
-                    bindMap.put("bool_value", null);
-                    bindMap.put("text_value", null);
-                  }
-
-                  logger.debug("{}", bindMap.values());
-                  bindVariableList.add(bindMap.values().toArray());
+                if (value.isValueNode() // scalar or explicit null
+                    || isEmptyObject(value)
+                    || isEmptyArray(value)) {
+                  ImmutablePair<Map<String, Object>, List<String>> result =
+                      convertJsonNodeToBoundVariables(
+                          value, parsingContext, db, path, key, patching);
+                  Map<String, Object> boundVariables = result.left;
+                  firstLevelKeys.addAll(result.right);
+                  logger.debug("{}", boundVariables.values());
```

Comment:
I know this is from legacy code but this seems like it's missing any context as to values and perhaps something that really should not be logged at any level? :)
But if it is maybe it should have some info on what those values are ("Bound variables from JSON: {}")

Reply:
Yeah, upon reading through it again it seems pretty silly to log this - I think waaaaay back in the day we had concerns that this code could be acting wrong and writing unexpected data to C* - but at this point we are pretty confident in its correctness, with testing and all. Removed!

Reply:
Yeah, it is likely it once served a purpose in figuring out things; nothing wrong with that.